modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/40_dhcp.js

```diff
@@ -127,29 +133,98 @@ return baseclass.extend({
 			else
 				exp = '%t'.format(lease.expires);
 
-			const hint = lease.macaddr ? machints.filter(function(h) { return h[0] == lease.macaddr })[0] : null;
+			const hint = lease.macaddr ? this.mac_hints.filter(function(h) { return h[0] == lease.macaddr })[0] : null;
+			let host = null;
+
+			if (hint && lease.hostname && lease.hostname != hint[1])
+				host = '%s (%s)'.format(lease.hostname, hint[1]);
+			else if (lease.hostname)
+				host = lease.hostname;
+
+			const vendor = this.ufp_list?.[lease.macaddr.toLowerCase()]?.vendor ?? null;
+			const mac_desc = vendor ? `${lease.macaddr} (${vendor})` : lease.macaddr;
+
+			const columns = [
+				host || '-',
+				lease.ipaddr,
+				mac_desc,
+				lease.duid || '-',
+				exp,
+			];
+
+			if (!this.isReadonlyView && lease.macaddr) {
+				columns.push(E('button', {
+					'class': 'cbi-button cbi-button-apply',
+					'click': L.bind(this.handleCreateStaticLease4, this, lease),
+					'data-tooltip': _('Reserve a specific IP address for this device'),
+					'disabled': this.isMACStatic[lease.macaddr.toUpperCase()]
+				}, [ _('Reserve IP') ]));
+			}
+
+			return columns;
+		}, this)), E('em', _('There are no active leases')));
+
+		return table4;
+	},
+
+	createOdhcpdTable4(iface_leases4) {
+		console.log("DHCPv4 leases " + JSON.stringify(iface_leases4));
+
+		const table4 = E('table', { 'id': 'status_leases4', 'class': 'table leases4' }, [
+			E('tr', { 'class': 'tr table-titles' }, [
+				E('th', { 'class': 'th' }, _('Interface')),
+				E('th', { 'class': 'th' }, _('Hostname')),
+				E('th', { 'class': 'th' }, _('IPv4 address')),
+				E('th', { 'class': 'th' }, _('MAC address')),
+				E('th', { 'class': 'th' }, _('FR')),
+				E('th', { 'class': 'th' }, _('Lease time remaining')),
+				this.isReadonlyView ? E([]) : E('th', { 'class': 'th cbi-section-actions' }, _('Static Lease'))
+			])
+		]);
+
+		const leases4 = Object.entries(iface_leases4).flatMap(([iface, { leases }]) =>
+			leases.map(lease => {
+				lease.iface = iface;
+				lease.reconf = lease['accept-reconf'];
+				lease.macaddr = lease.mac.toUpperCase().match(/.{1,2}/g).join(':');
+				lease.ipaddr = lease.address;
+				return lease;
+			})
+		);
+
+		cbi_update_table(table4, leases4.map(L.bind(function(lease) {
+			let exp;
+			if (lease.valid == 0xffffffff)
+				exp = E('em', _('unlimited'));
+			else if (lease.valid <= 0)
+				exp = E('em', _('expired'));
+			else
+				exp = '%t'.format(lease.valid);
```

Comment:
just use interpolated string.

Reply:
How do I do that for `%t`? That's a `cbi.js` addition to the String class, isn't it?